pkg/apis/triggers/v1alpha1/event_listener_validation_test.go

```diff
@@ -4,26 +4,84 @@ import (
 	"context"
 	"testing"
 
-	v1alpha1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
 	bldr "github.com/tektoncd/triggers/test/builder"
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	fakedynamicclient "k8s.io/client-go/dynamic/fake"
 )
 
 func Test_EventListenerValidate(t *testing.T) {
 	tests := []struct {
-		name     string
-		el       *v1alpha1.EventListener
-		raiseErr bool
+		name string
+		el   *v1alpha1.EventListener
 	}{{
 		name: "TriggerTemplate Does Not Exist",
 		el: bldr.EventListener("name", "namespace",
 			bldr.EventListenerSpec(
 				bldr.EventListenerTrigger("tb", "dne", "v1alpha1"))),
-		raiseErr: true,
 	}, {
+		name: "Valid EventListener No TriggerBinding",
+		el: bldr.EventListener("name", "namespace",
+			bldr.EventListenerSpec(
+				bldr.EventListenerTrigger("", "tt", "v1alpha1"))),
+	}, {
+		name: "Valid EventListener No Interceptor",
+		el: bldr.EventListener("name", "namespace",
+			bldr.EventListenerSpec(
+				bldr.EventListenerTrigger("tb", "tt", "v1alpha1"))),
+	}, {
+		name: "Valid EventListener Interceptor Name only",
+		el: bldr.EventListener("name", "namespace",
+			bldr.EventListenerSpec(
+				bldr.EventListenerTrigger("tb", "tt", "v1alpha1",
+					bldr.EventListenerTriggerInterceptor("svc", "", "", "")))),
```

Comment:
im confused why it is valid to only specify the name of an interceptor, is that enough to be able to address it?

Reply:
Good catch. Not sure why the validation was done this way but we did return an error later when we actually try to resolve it to a URL.

Reply:
Made this an error test!

Reply:
wooooot :D